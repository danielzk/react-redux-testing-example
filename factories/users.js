import {Factory} from 'rosie'
import {OrderedMap} from 'immutable'

export const UserFactory = new Factory()
  .sequence('id', sequence => sequence)
  .sequence('username', sequence => 'username' + sequence)
  .attr('firstName', 'Mark')
  .attr('lastName', 'Zucaritas')

export const UserListFactory = (count) => {
  const users = UserFactory.buildList(count)
  return OrderedMap(users.map(item => [item.id, item]))
}
