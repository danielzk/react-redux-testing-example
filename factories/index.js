import {Factory} from 'rosie'

import {UserFactory, UserListFactory} from './users'

export const StoreFactory = new Factory()
  .option('numUsers', 2)
  .attr('users', ['numUsers'], count => UserListFactory(count))

export {UserFactory, UserListFactory}
