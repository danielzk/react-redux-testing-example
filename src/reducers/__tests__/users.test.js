import reducer from '../users'
import types from '../../actionTypes'
import {UserFactory, UserListFactory} from '../../../factories'

describe('Users reducer', () => {
  it('should handle initial state', () => {
    const state = reducer(undefined, {})
    expect(state).toMatchSnapshot()
  })

  it('ADD_USER should add user', () => {
    const user = UserFactory.build()
    const state = reducer(undefined, {type: types.ADD_USER, payload: {user}})
    expect(state).toMatchSnapshot()
  })

  it('REMOVE_USER should remove user', () => {
    const users = UserListFactory(2)
    const state = reducer(users, {type: types.REMOVE_USER, payload: {id: users.get(1).id}})
    expect(state).toMatchSnapshot()
  })

  it('SET_USERS should set users', () => {
    const users = UserListFactory(2)
    const state = reducer(undefined, {type: types.SET_USERS, payload: {users}})
    expect(state).toMatchSnapshot()
  })
})
