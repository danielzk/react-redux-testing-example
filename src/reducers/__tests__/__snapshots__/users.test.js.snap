// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users reducer ADD_USER should add user 1`] = `
Immutable.OrderedMap {
  1: Object {
    "firstName": "Mark",
    "id": 1,
    "lastName": "Zucaritas",
    "username": "username1",
  },
}
`;

exports[`Users reducer REMOVE_USER should remove user 1`] = `
Immutable.OrderedMap {
  2: Object {
    "firstName": "Mark",
    "id": 2,
    "lastName": "Zucaritas",
    "username": "username2",
  },
}
`;

exports[`Users reducer SET_USERS should set users 1`] = `
Immutable.OrderedMap {
  1: Object {
    "firstName": "Mark",
    "id": 1,
    "lastName": "Zucaritas",
    "username": "username1",
  },
  2: Object {
    "firstName": "Mark",
    "id": 2,
    "lastName": "Zucaritas",
    "username": "username2",
  },
}
`;

exports[`Users reducer should handle initial state 1`] = `Immutable.OrderedMap {}`;
